### PR TITLE
blog post for v0.31.0: github IDP support

### PR DESCRIPTION
**Pull request by cfryanr**
Preview the blog post here: https://deploy-preview-1980--pinniped.netlify.app/posts/github-idp-support/

**Release note**:

None, docs-only change.

```release-note
NONE
```

### CODE REVIEW

File: site/content/posts/2024-06-05-githubidentityprovider.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        - my-enterprise-organization # this is case-insensitive
+  client:
+    secretName: my-github-provider-client-secret
+```
+
+Now users must belong to the organization configured above to successfully authenticate.
+
+When multiple orgs are `allowed` then the user must belong to any one of those orgs.
+
+Want to further restrict auth by GitHub team membership? No problem, you can still use the technique shown above.
```
Comment (joshuatcasey) [inline site/content/posts/2024-06-05-githubidentityprovider.md:124]: Clarify what `technique shown above` means?

Comment (joshuatcasey) [reply site/content/posts/2024-06-05-githubidentityprovider.md:124]: I know it means the `policy/v1` expression above but we should clarify.

File: site/content/posts/2024-06-05-githubidentityprovider.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+`.spec.claims.username` allows you to choose from:
+- `login`: The user's GitHub username as shown on their profile. Note that a user can change their own username,
+   so this is not recommended for production use with identities from public github.com.
+- `id`: The numeric user ID assigned by GitHub will be used as the username. On public github.com, this can be found for any user
+   by putting their login name into this GitHub API URL: `https://api.github.com/users/cfryanr` (replace `cfryanr` with the login name).
+   This is automatically assigned and immutable for each user.
+- `login:id`: Blends the readability of using login names with the immutability of using IDs by putting both into
+   the Kubernetes usernames, separated by a colon. This keeps your RBAC policies nicely readable.
+
+`spec.claims.groups` allows you to choose from:
```
Comment (joshuatcasey) [inline site/content/posts/2024-06-05-githubidentityprovider.md:163]: `.spec` like above?

File: site/content/posts/2024-06-05-githubidentityprovider.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+      policy: OnlyUsersFromAllowedOrganizations
+      allowed:
+        - my-enterprise-organization
+  client:
+    secretName: my-github-provider-client-secret
+```
+
+## Kubernetes username and group names
+
+The GitHubIdentityProvider resource offers several choices for how your users' Kubernetes usernames and group names should look.
```
Comment (joshuatcasey) [inline site/content/posts/2024-06-05-githubidentityprovider.md:152]: Should we say what the defaults are?

File: site/content/posts/2024-06-05-githubidentityprovider.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+Your Kubernetes RBAC policies will see the updated group memberships after the next refresh.
+There is no way for your end users to avoid these refreshes without losing access to your clusters.
+
+## Where to read more
+
+This blog post is just a quick overview of this new feature. To learn about how to configure the Pinniped Supervisor
+with this new feature, see:
+
+- The [GitHub configuration guide]({{< ref "docs/howto/supervisor/configure-supervisor-with-github.md" >}}).
+- The [GitHubIdentityProvider resource](https://github.com/vmware-tanzu/pinniped/blob/main/generated/latest/README.adoc#k8s-api-go-pinniped-dev-generated-1-30-apis-supervisor-idp-v1alpha1-githubidentityprovider) documentation.
```
Comment (joshuatcasey) [inline site/content/posts/2024-06-05-githubidentityprovider.md:188]: Is this anchor permanent? I see `latest` in the path (which seems correct) but `1-30` in the anchor.

File: site/content/posts/2024-06-05-githubidentityprovider.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
+---
+title: "Pinniped v0.31.0: GitHub as an identity provider"
+slug: github-idp-support
+date: 2024-06-05
+author: Ryan Richard
+image: https://images.unsplash.com/photo-1557657043-23eec69b89c9?q=80&w=3008&auto=format&fit=crop&ixlib=rb-4.0.3
+excerpt: "With the release of v0.31.0, Pinniped brings GitHub identities to Kubernetes clusters everywhere"
+tags: ['Ryan Richard', 'release']
```
Comment (joshuatcasey) [inline site/content/posts/2024-06-05-githubidentityprovider.md:8]: Should we tag all the contributors?

File: site/content/posts/2024-06-05-githubidentityprovider.md (hunk 1 above)
Comment (cfryanr) [reply site/content/posts/2024-06-05-githubidentityprovider.md]: Updated and force pushed.

File: site/content/posts/2024-06-05-githubidentityprovider.md (hunk 2 above)
Comment (cfryanr) [reply site/content/posts/2024-06-05-githubidentityprovider.md:164]: Updated and force pushed. Actually removed the leading `.` from the other one.

File: site/content/posts/2024-06-05-githubidentityprovider.md (hunk 3 above)
Comment (cfryanr) [reply site/content/posts/2024-06-05-githubidentityprovider.md:153]: Updated and force pushed.

File: site/content/posts/2024-06-05-githubidentityprovider.md [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+`spec.claims.groups` allows you to choose from:
+- `name`: GitHub team names can include mixed-case characters, spaces, and punctuation, e.g. `Kube admins!`.
+- `slug`: GitHub slug names are lower-cased, with spaces replaced by hyphens, and other punctuation removed, e.g. `kube-admins`.
+- Either way, the team names will automatically be prefixed by the name of the org in which the team resides, with a `/` separator,
+  e.g. `My-org/kube-admins`. The org name will preserve its case from GitHub.
+
+## Control new and existing sessions by changing org and team memberships on GitHub
+
+Did one of your developers or operators just change teams or leave your enterprise? Fear not. Simply update their
+GitHub organization and/or GitHub team memberships on github.com and Pinniped will respect those changes immediately.
```
[the remark this answers is not in the document]
Comment (cfryanr) [reply site/content/posts/2024-06-05-githubidentityprovider.md]: Updated and force pushed.

File: site/content/posts/2024-06-05-githubidentityprovider.md (hunk 4 above)
Comment (cfryanr) [reply site/content/posts/2024-06-05-githubidentityprovider.md]: Updated and force pushed. Good catch.

File: site/content/posts/2024-06-05-githubidentityprovider.md (hunk 5 above)
Comment (cfryanr) [reply site/content/posts/2024-06-05-githubidentityprovider.md:8]: The tag only identifies the author of the blog post, not the authors/contributors of the code. It is not shown in the rendered blog post. It only decides what should be rendered on this page which you get to by clicking on my name: https://deploy-preview-1980--pinniped.netlify.app/tags/ryan-richard/

**Review by joshuatcasey — APPROVED** [review]

**Pull request merged**